app/src/main/java/com/github/se/cyrcle/io/serializer/ParkingAdapter.kt

```diff
@@ -12,37 +17,50 @@ import com.google.gson.reflect.TypeToken
 import com.mapbox.geojson.Point
 import java.lang.reflect.Type
 
-/** Adapter for serializing and deserializing Parking objects. */
+/**
+ * Adapter for serializing and deserializing Parking objects. Functions annotated
+ * with @TypeConverter are used by Room to convert the data type of a field in the database.
+ */
 class ParkingAdapter : JsonSerializer<Parking>, JsonDeserializer<Parking> {
 
   /** GSON instance with a custom adapter for Point objects. */
-  private val gson = GsonBuilder().registerTypeAdapter(Point::class.java, PointAdapter()).create()
+  private val gson =
+      GsonBuilder()
+          .registerTypeAdapter(Point::class.java, PointAdapter())
+          .registerTypeAdapter(Location::class.java, LocationAdapter())
+          .create()
 
   /** Type token for a map of Strings to Objects. */
   private val serializedParkingType = object : TypeToken<Map<String, Any>>() {}.type
 
-  /**
-   * Serializes a Parking object to a JSON string. Used internally by Room when this class is
-   * declared as a TypeConverter.
-   *
-   * @param parking The Parking object to serialize.
-   * @return The serialized JSON string.
-   */
   @TypeConverter
-  fun serialize(parking: Parking): String {
-    return gson.toJson(parking)
+  fun serializeParkingCapacity(parkingCapacity: ParkingCapacity): Int {
+    return parkingCapacity.ordinal
+  }
+
+  @TypeConverter
+  fun deserializeParkingCapacity(ordinal: Int): ParkingCapacity {
```

Comment:
Yeah I definitely agree w/ @andrepeiry here. If we're dealing w/ indices in a table we need to error handle... it'd be quite a shame for the app to consistenly crash if we make a change at this level